File: py/data_structures/utils/LinkedList.py
class DoublyLinkedListNode:
    """
    This class represents a node meant to be used within doubly linked lists.
    """ 
    def __init__(self, val):
        self.val = val
        self.next = None
        self.prev = None 

class SinglyLinkedListNode:
    """
    This class represents a node meant to be used within singly linked lists.
    """ 
    def __init__(self, val):
        self.val = val
        self.next = None

File: js/data structures/utils/LinkedList.js
/**
 * This class represents a doubly linked node, where every node has two pointers. Meant to be used within
 * a doubly linked list. 
 * @class @public 
 */
class DoubleLinkedListNode {
    constructor(val) {
        this.val = val;
        this.prev = null;
        this.next = null;
    }
}
/**
 * This class represents a singly linked node, where every node has one pointer. Meant to be used within a
 * singly linked list. 
 * @class @public 
 */
 class SingleLinkedListNode {
    constructor(val) {
        this.val = val;
        this.next = null;
    }
}

export {DoubleLinkedListNode, SingleLinkedListNode}; 